RecoBTag/PerformanceDB/test/TestBtagPayloads.cc

```diff
@@ -36,19 +36,94 @@
+  //                     MUJETSWPBTAGJPL,   MUJETSWPBTAGJPM,    MUJETSWPBTAGJPT
+  //                                                            MUJETSWPBTAGTCHPT
+  constexpr char const* const muName = "MUJETSWPBTAGCSVL";
+
+  //Possible algorithms: MISTAGCSVLAB,  MISTAGCSVMAB,   MISTAGCSVTAB
+  //                     MISTAGJPLAB,   MISTAGJPMAB,    MISTAGJPTAB                // Data period 2012 AB
+  //                                                    MISTAGTCHPTAB
+  //
+  //                     MISTAGCSVLABCD,  MISTAGCSVMABCD,   MISTAGCSVTABCD
+  //                     MISTAGJPLABCD,   MISTAGJPMABCD,    MISTAGJPTABCD          // Data period 2012 ABCD
+  //                                                        MISTAGTCHPTABCD
+  //
+  //                     MISTAGCSVLC,  MISTAGCSVMC,   MISTAGCSVTC
+  //                     MISTAGJPLC,   MISTAGJPMC,    MISTAGJPTC                   // Data period 2012 C
+  //                                                  MISTAGTCHPTC
+  //
+  //                     MISTAGCSVLD,  MISTAGCSVMD,   MISTAGCSVTD
+  //                     MISTAGJPLD,   MISTAGJPMD,    MISTAGJPTD                   // Data period 2012 D
+  //                                                  MISTAGTCHPTD
+  constexpr char const* const mistagName = "MISTAGTCHPTD";
+}  // namespace
+TestBtagPayloads::TestBtagPayloads(const edm::ParameterSet& iConfig)
+    : ttToken_(esConsumes(edm::ESInputTag("", ttName))),
+      muToken_(esConsumes(edm::ESInputTag("", muName))),
+      mistagToken_(esConsumes(edm::ESInputTag("", mistagName))),
+      measureName_({"TTBARWPBTAGCSVL",
+                    "TTBARWPBTAGCSVL",
+                    "TTBARWPBTAGCSVL",
+                    "TTBARWPBTAGCSVL",
+                    "TTBARWPBTAGJPT",
+                    "TTBARWPBTAGJPT",
+                    "TTBARWPBTAGJPT",
+                    "TTBARWPBTAGJPT"}),
+      measureType_({"BTAGBEFFCORR",
+                    "BTAGBERRCORR",
+                    "BTAGCEFFCORR",
+                    "BTAGCERRCORR",
+                    "BTAGBEFFCORR",
+                    "BTAGBERRCORR",
+                    "BTAGCEFFCORR",
+                    "BTAGCERRCORR"}) {
+  if (measureName_.size() != measureType_.size()) {
+    std::cout << "measureName_, measureType_ size mismatch!" << std::endl;
+    exit(-1);
+  }
+
+  measureMap_["BTAGBEFFCORR"] = PerformanceResult::BTAGBEFFCORR;
+  measureMap_["BTAGBERRCORR"] = PerformanceResult::BTAGBERRCORR;
+  measureMap_["BTAGCEFFCORR"] = PerformanceResult::BTAGCEFFCORR;
+  measureMap_["BTAGCERRCORR"] = PerformanceResult::BTAGCERRCORR;
+
+  measureTokens_.reserve(measureName_.size());
+
+  //only call esConsumes if the name changes, else reuse the token
+  std::string lastName;
+  edm::ESGetToken<BtagPerformance, BTagPerformanceRecord> lastToken;
+  for (auto const& n : measureName_) {
+    if (lastName != n) {
+      lastName = n;
+      lastToken = esConsumes(edm::ESInputTag("", n));
```

Comment:
is there some cost from calling `esConsumes` multiple times and more importantly, is there a change in logic if it's called multiple times?
- If there is no harm, then this optimization to skip is ok, 
- if the logic changes, then this check seems rather weak (assuming that `measureName_` is sorted, as in the current hardcoded implementation) and some more thorough check is needed (perhaps the easiest is to make a sorted temporary)

Reply:
There is only a minor performance hit.
- needs more memory to store the consumes info
- each identical consumes will create an additional entry in the WaitingTaskHolder for the item which takes more memory and a small bit of time.

Mostly, this bugs me aesthetically.